xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1254,177 +1242,42 @@ HDR_STATUS CWIN32Util::ToggleWindowsHDR(DXGI_MODE_DESC& modeDesc)
     }
   }
 #else
-  uint32_t pathCount = 0;
-  uint32_t modeCount = 0;
+  std::wstring gdiDeviceName{CDisplayUtilsWin32::GetCurrentDisplayName()};
+  if (gdiDeviceName.empty())
+    return HDR_STATUS::HDR_TOGGLE_FAILED;
 
-  MONITORINFOEXW mi = {};
-  mi.cbSize = sizeof(mi);
-  GetMonitorInfoW(MonitorFromWindow(g_hWnd, MONITOR_DEFAULTTOPRIMARY), &mi);
-  const std::wstring deviceNameW = mi.szDevice;
+  // The graphics mode is altered by the Windows HDR status change, save it before
+  DEVMODEW devMode = {};
+  devMode.dmSize = sizeof(devMode);
 
-  if (ERROR_SUCCESS != GetDisplayConfigBufferSizes(QDC_ONLY_ACTIVE_PATHS, &pathCount, &modeCount))
-    return status;
+  if (!EnumDisplaySettingsW(gdiDeviceName.c_str(), ENUM_CURRENT_SETTINGS, &devMode))
```

Comment:
```suggestion
  if (FALSE == EnumDisplaySettingsW(gdiDeviceName.c_str(), ENUM_CURRENT_SETTINGS, &devMode))
```
EnumDisplaySettingsW retuns type BOOL, not bool

Reply:
yes